File: demo/src/main/java/com/devbrackets/android/exomediademo/data/MediaItem.java
package com.devbrackets.android.exomediademo.data;

import com.devbrackets.android.exomediademo.manager.PlaylistManager;
import com.devbrackets.android.playlistcore.annotation.SupportedMediaType;
import com.devbrackets.android.playlistcore.api.PlaylistItem;

/**
 * A custom {@link PlaylistItem}
 * to hold the information pertaining to the audio and video items
 */
public class MediaItem implements PlaylistItem {
    private Samples.Sample sample;
    boolean isAudio;

    public MediaItem(Samples.Sample sample, boolean isAudio) {
        this.sample = sample;
        this.isAudio = isAudio;
    }

    @Override
    public long getId() {
        return 0;
    }

    @Override
    public boolean getDownloaded() {
        return false;
    }

    @Override
    @SupportedMediaType
    public int getMediaType() {
        return isAudio ? PlaylistManager.AUDIO : PlaylistManager.VIDEO;
    }

    @Override
    public String getMediaUrl() {
        return sample.getMediaUrl();
    }

    @Override
    public String getDownloadedMediaUri() {
        return null;
    }

    @Override
    public String getThumbnailUrl() {
        return sample.getArtworkUrl();
    }

    @Override
    public String getArtworkUrl() {
        return sample.getArtworkUrl();
    }

    @Override
    public String getTitle() {
        return sample.getTitle();
    }

    @Override
    public String getAlbum() {
        return "PlaylistCore Demo";
    }

    @Override
    public String getArtist() {
        return "Unknown Artist";
    }
}

File: demo/src/main/kotlin/com/devbrackets/android/exomediademo/data/MediaItem.kt
package com.devbrackets.android.exomediademo.data

import com.devbrackets.android.playlistcore.annotation.SupportedMediaType
import com.devbrackets.android.playlistcore.api.PlaylistItem
import com.devbrackets.android.playlistcore.manager.BasePlaylistManager

/**
 * A custom [PlaylistItem]
 * to hold the information pertaining to the audio and video items
 */
class MediaItem(private val sample: Samples.Sample, internal var isAudio: Boolean) : PlaylistItem {

    override val id: Long
        get() = 0

    override val downloaded: Boolean
        get() = false

    @SupportedMediaType
    override val mediaType: Int
        get() = if (isAudio) BasePlaylistManager.AUDIO else BasePlaylistManager.VIDEO

    override val mediaUrl: String?
        get() = sample.mediaUrl

    override val downloadedMediaUri: String?
        get() = null

    override val thumbnailUrl: String?
        get() = sample.artworkUrl

    override val artworkUrl: String?
        get() = sample.artworkUrl

    override val title: String?
        get() = sample.title

    override val album: String?
        get() = "PlaylistCore Demo"

    override val artist: String?
        get() = "Unknown Artist"
}